rotkehlchen/history/events/structures/swap.py

```diff
@@ -139,24 +139,25 @@ def deserialize_from_db(cls: type['SwapEvent'], entry: tuple) -> 'SwapEvent':
             notes=entry[8] or None,
         )
 
-    def _generate_auto_notes(self) -> str:
+    def _generate_auto_notes(self, location: str) -> str:
```

Comment:
you need to explain the new argument and what it does in the docstring and why nto always use self.location